Fix markdown for the issue and PR templates

papandreou opened the pull request:
So that the checkboxes are rendered correctly.

<!--
    Thank you for contributing!

    ESLint adheres to the [JS Foundation Code of Conduct](https://js.foundation/community/code-of-conduct).
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).
- [ ] The team has reached consensus on the changes proposed in this pull request. If not, I understand that the evaluation process will begin with this pull request and won't be merged until the team has reached consensus.

#### What is the purpose of this pull request? (put an "X" next to an item)

- [ ] Documentation update
- [ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
- [ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
- [ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
- [ ] Add autofixing to a rule
- [ ] Add a CLI option
- [ ] Add something to the core
- [x] Other, please explain:

<!--
    If the item you've checked above has a template, please paste the template questions below and answer them. (If this pull request is addressing an issue, you can just paste a link to the issue here instead.)
-->

<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/developer-guide/contributing/pull-requests)
    - Include tests for this change
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

#### What changes did you make? (Give an overview)

When opening another PR I noticed that a bunch of lines starting with `[ ]` that came from the github PR template weren't properly rendered as a list of checkboxes. I changed the templates so they do :hugs:

kaicataldo commented:
Thanks for contributing. The current template is that way by design, because if they were actual checkboxes the PR list view would show 1/x checkboxes checked. It would be nice if we had some way of creating radio buttons that only allowed one to be selected, but markdown unfortunately doesn’t support that.

papandreou commented:
Haha, okay, that's fair enough. It seems like contributors are confused enough that they only follow the instructions half of the time, though. I went through the first 4 pages of PRs and counted:

* 47 left the list as-is, added one `X` in one of the boxes :ok_hand:
* 39 edited the list down to just one item with `[X]` in front of it
* 5 left the list as-is, added `X` in two of the boxes
* 5 edited the markdown so it's a list of checkboxes, then added one `X` (this includes my PRs)
* 1 edited the list down to one item without `[X]`
* 1 edited the list down to two items with `[X]` in front
* 1 left the list as-is without adding any `X` at all :thinking:
* 1 removed the list altogether :see_no_evil:

I guess it doesn't really matter, actually. But I'll maintain that it does look like a mistake in the templates when you know the gh flavored markdown syntax for a checkbox :smirk_cat:

kaicataldo commented:
Agreed. I think there are two issues at play:

1. The one you mentioned, where it can easily seem like a typo if you’re used to writing lists in markdown.
2. We as a team haven’t been very good at enforcing these rules.

It definitely would be nice to figure out a better system. In my personal experience, templates in general don’t seem to be very effective in issues or PRs (though they’re better than nothing!).

kaicataldo commented:
If you have a suggestion for an alternative system, I’d love to hear your idea!

nzakas commented:
Agree with @kaicataldo that the template format is intentional. I’d suggest opening an issue if you have other ideas about how to improve the template.
